src/app/api/file/route.ts

```diff
@@ -0,0 +1,362 @@
+import { NextResponse } from "next/server";
+import { File, FileResponse } from "./route.schema";
+import { google } from "googleapis";
+import { prisma } from "@server/db/client";
+import { withSession } from "../../../server/decorator";
+import { env } from "../../../env/server.mjs";
+
+const initializeDriveAuth = async (request) => {
+  const { access_token, refresh_token } = (await prisma.account.findFirst({
+    where: {
+      userId: request.session.user.id,
+    },
+  })) ?? { access_token: null };
+
+  const auth = new google.auth.OAuth2({
+    clientId: env.GOOGLE_CLIENT_ID,
+    clientSecret: env.GOOGLE_CLIENT_SECRET,
+  });
+
+  auth.setCredentials({
+    access_token,
+    refresh_token,
+  });
+
+  const service = google.drive({
+    version: "v3",
+    auth,
+  });
+
+  const body = await request.req.json();
+  const fileRequest = File.safeParse(body);
+
+  return { service, fileRequest };
+};
+
+export const POST = withSession(async (request) => {
+  try {
+    const { service, fileRequest } = await initializeDriveAuth(request);
+
+    if (!fileRequest.success) {
+      console.log(fileRequest.error);
+      return NextResponse.json(
+        FileResponse.parse({
+          code: "NOT_FOUND",
+          message: "Unsuccessful request creation",
+        }),
+        { status: 400 }
+      );
+    } else {
+      const fileData = fileRequest.data;
+
+      // Check that user has this senior assigned to them
+      const { SeniorIDs } = await prisma.user.findFirst({
+        where: {
+          id: request.session.user.id,
+        },
+      });
+
+      if (
+        !SeniorIDs.some((seniorId: string) => seniorId === fileData.seniorId)
+      ) {
+        return NextResponse.json(
+          FileResponse.parse({
+            code: "NOT_AUTHORIZED",
+            message: "Senior not assigned to user",
+          }),
+          { status: 404 }
+        );
+      }
+
+      // Get senior from database
+      const foundSenior = await prisma.senior.findUnique({
+        where: { id: fileData.seniorId },
+      });
+      if (foundSenior == null) {
+        return NextResponse.json(
+          FileResponse.parse({
+            code: "NOT_FOUND",
+            message: "Senior not found",
+          }),
+          { status: 404 }
+        );
+      }
+
+      const parentID = foundSenior.folder.split("/").pop();
+      const formatted_date =
```

Comment:
I'm a bit confused by this logic. Why can't we pass in `new Date()` directly? https://www.prisma.io/dataguide/mongodb/working-with-dates#how-to-create-new-dates

Reply:
It's cause we want the title to be in the format of MM/DD/YYYY but if we just do toISOString or just pass in the Date the format's a really long string